docs/content/querying/dimensionspecs.md

```diff
@@ -15,9 +15,18 @@ The following JSON fields can be used in a query to operate on dimension values.
 Returns dimension values as is and optionally renames the dimension.
 
 ```json
-{ "type" : "default", "dimension" : <dimension>, "outputName": <output_name> }
+{
+  "type" : "default",
+  "dimension" : <dimension>,
+  "outputName": <output_name>,
+  "outputType": <"STRING"|"LONG"|"FLOAT">
+}
 ```
 
+When specifying a DimensionSpec on a numeric column, the user should include the type of the column in the `outputType` field. This is necessary as it is possible for a column with given name to have different value types in different segments. Result merging may fail unless results of different type are converted to the type specified by `outputType`.
```

Comment:
I reworded this area to say that results will be converted to the type specified by outputType before merging